components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java

```diff
@@ -348,6 +350,46 @@ public static Tenant initializeTenant(TenantInfoBean tenantInfoBean) {
         return tenant;
     }
 
+    public static void updateTenantPassword(TenantInfoBean tenantInfoBean, String adminPassword) throws Exception {
+
+        if (StringUtils.isBlank(adminPassword)) {
+            return;
+        }
+        UserStoreManager userStoreManager;
+        try {
+            userStoreManager = getUserStoreManager(tenantInfoBean.getTenantId());
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            String msg = "Error in getting the user store manager for tenant, tenant domain: " +
+                    tenantInfoBean.getTenantDomain() + ".";
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new Exception(msg, e);
+        }
+        if (!userStoreManager.isReadOnly()) {
+            try {
+                userStoreManager.updateCredentialByAdmin(tenantInfoBean.getAdmin(), adminPassword);
+            } catch (UserStoreException e) {
+                String msg = String.format("Error in changing the tenant admin password for tenant domain: %s. %s",
+                        tenantInfoBean.getTenantDomain(), e.getMessage());
+                if (log.isDebugEnabled()) {
+                    log.debug(msg, e);
+                }
+                throw new UserStoreException(msg, e);
+            }
+        } else {
+            // Password should be empty since no password update done.
+            tenantInfoBean.setAdminPassword("");
```

Comment:
If the user store manager is readonly why do we need to update the bean to empty ? 
Why cannot we simply do nothing here ?

Reply:
Removed with https://github.com/wso2/carbon-multitenancy/pull/276/commits/0043c2e0afe9e14b5b41220933a614a619f62005